downloader/src/test/kotlin/DownloaderTest.kt

```diff
@@ -0,0 +1,145 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.downloader
+
+import com.here.ort.model.Package
+
+import io.kotlintest.Spec
+import io.kotlintest.matchers.beGreaterThan
+import io.kotlintest.matchers.contain
+import io.kotlintest.matchers.should
+import io.kotlintest.matchers.shouldBe
+import io.kotlintest.matchers.shouldNotBe
+import io.kotlintest.specs.StringSpec
+
+import java.io.File
+
+class DownloaderTest : StringSpec() {
+
+    private val outputDir = createTempDir()
+
+    override fun interceptSpec(context: Spec, spec: () -> Unit) {
+        spec()
+        outputDir.deleteRecursively()
+    }
+
+    init {
+        "Downloads gradle module by vcs path" {
+            val submoduleName = "model"
+            val pkg = Package(
+                    packageManager = "Gradle",
+                    namespace = "",
+                    name = "project :model",
+                    version = "",
+                    declaredLicenses = sortedSetOf(),
+                    description = "",
+                    homepageUrl = "",
+                    downloadUrl = "",
+                    hash = "",
+                    hashAlgorithm = "",
+                    vcsPath = submoduleName,
+                    vcsProvider = "Git",
+                    vcsUrl = "git@github.com:heremaps/oss-review-toolkit.git",
+                    vcsRevision = ""
+            )
+            Main.download(pkg, outputDir)
+            val downloadedProjectDir = File(outputDir, pkg.normalizedName)
+            val downloadedModuleDir = File(downloadedProjectDir, submoduleName)
+
+            outputDir.list().size shouldBe 1
+            outputDir.list().first() shouldBe pkg.normalizedName
+
+            downloadedProjectDir.isDirectory shouldBe true
+            downloadedModuleDir.isDirectory shouldBe true
+
+            downloadedModuleDir.list().size shouldBe beGreaterThan(1)
+
+            val moduleDirList = downloadedModuleDir.list().asList()
+            moduleDirList should contain("build.gradle")
+            moduleDirList should contain("src")
+        }.config(tags = setOf(IntegrationTest))
+
+        "Downloads github oss-review-toolkit repo " {
+            val pkg = Package(
+                    packageManager = "Gradle",
+                    namespace = "",
+                    name = "oss-review-toolkit",
+                    version = "",
+                    declaredLicenses = sortedSetOf(),
+                    description = "",
+                    homepageUrl = "",
+                    downloadUrl = "",
+                    hash = "",
+                    hashAlgorithm = "",
+                    vcsPath = "",
```

Comment:
We meanwhile swapped arguments a bit, please make `vcsPath` go after `vcsRevision`.